### PR TITLE
crimson/os/seastore: handle OP_COLL_MOVE_RENAME

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
jenkins test api

File: src/crimson/os/seastore/seastore.cc [hunk 1]
```diff
@@ -1479,6 +1495,37 @@ SeaStore::Shard::_do_transaction_step(
...
+  auto object_data = olayout.object_data.get();
+  auto oi_bl = ceph::bufferlist::static_from_mem(
+    &olayout.oi[0],
+    (uint32_t)olayout.oi_size);
+
+  d_onode->update_onode_size(*ctx.transaction, size);
+  d_onode->update_omap_root(*ctx.transaction, omap_root);
+  d_onode->update_xattr_root(*ctx.transaction, xattr_root);
+  d_onode->update_object_data(*ctx.transaction, object_data);
+  d_onode->update_object_info(*ctx.transaction, oi_bl);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/seastore.cc:1519]: Looks snapset is missed?

Comment (xxhdx1985126) [reply src/crimson/os/seastore/seastore.cc:1547]: Added.

File: src/crimson/os/seastore/seastore.cc [hunk 2]
```diff
@@ -1504,6 +1520,41 @@ SeaStore::Shard::_do_transaction_step(
...
+  d_onode->update_snapset(*ctx.transaction, ss_bl);
+  return onode_manager->erase_onode(
+    *ctx.transaction, onode
+  ).handle_error_interruptible(
+    crimson::ct_error::input_output_error::pass_further(),
+    crimson::ct_error::assert_all{
+      "Invalid error in SeaStore::_remove"}
```
Comment (cyx1231st) [inline src/crimson/os/seastore/seastore.cc:1554]: _remove -> _rename

**Review by cyx1231st — APPROVED** [review]

File: src/crimson/os/seastore/seastore.cc (hunk 2 above)
Comment (xxhdx1985126) [reply src/crimson/os/seastore/seastore.cc]: Done:-)

**Comment by cyx1231st** [comment]
jenkins test make check

**Comment by cyx1231st** [comment]
Waiting for https://shaman.ceph.com/builds/ceph/ci-yingxin-crimson-pr-55947/

**Comment by cyx1231st** [comment]
jenkins test make check

**Comment by cyx1231st** [comment]
jenkins test make check

**Pull request merged**